packages/medusa/src/api-v2/store/product-categories/helpers.ts

```diff
@@ -0,0 +1,38 @@
+import { MedusaContainer } from "@medusajs/types"
+import {
+  ContainerRegistrationKeys,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+
+export const refetchCategory = async (
+  categoryId: string,
+  scope: MedusaContainer,
+  fields: string[],
+  filterableFields: Record<string, any> = {}
+) => {
+  const remoteQuery = scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+  const queryObject = remoteQueryObjectFromString({
+    entryPoint: "product_category",
+    variables: {
+      filters: { ...filterableFields, id: categoryId },
+    },
+    fields: fields,
+  })
+
+  const categories = await remoteQuery(queryObject)
+  return categories[0]
+}
+
+export const applyCategoryFilters = (req, res, next) => {
```

Comment:
I've added this middleware you can now use - https://github.com/medusajs/medusa/pull/7178/files#diff-174c1cb2d4fa4db31b18bbb3956cf29a21cfccf26fcaecdc2f9dacce55143484

Reply:
@riqwan I'll do a follow-up cleanup PR, I just want to get things merged before we do the v1 cleanup.